bundles/org.openhab.binding.teslascope/src/main/java/org/openhab/binding/teslascope/internal/TeslascopeHandler.java

```diff
@@ -0,0 +1,381 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.teslascope.internal;
+
+import static org.openhab.binding.teslascope.internal.TeslascopeBindingConstants.*;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.teslascope.internal.api.DetailedInformation;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.ImperialUnits;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TeslascopeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Paul Smedley - Initial contribution
+ */
+@NonNullByDefault
+public class TeslascopeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(TeslascopeHandler.class);
+
+    private @NonNullByDefault({}) TeslascopeConfiguration config;
+    private @NonNullByDefault({}) TeslascopeWebTargets webTargets;
+    private @Nullable ScheduledFuture<?> pollFuture;
+
+    public TeslascopeHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        webTargets = new TeslascopeWebTargets(httpClient);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            switch (channelUID.getId()) {
+                case TeslascopeBindingConstants.CHANNEL_AUTOCONDITIONING:
+                    if (command instanceof OnOffType onOffCommand) {
+                        if (onOffCommand == OnOffType.ON) {
+                            ac(true);
+                        } else {
+                            ac(false);
+                        }
```

Comment:
This suggestion can also be applied to the other cases in this block.
```suggestion
                        ac(onOffCommand == OnOffType.ON);
```

Reply:
Nice! Done for all instances.